flink-libraries/flink-state-processing-api/src/main/java/org/apache/flink/state/api/output/SnapshotUtils.java

```diff
@@ -57,7 +58,7 @@ private SnapshotUtils() {}
 
         CheckpointOptions options =
                 CheckpointOptions.forConfig(
-                        SavepointType.savepoint(SavepointFormatType.CANONICAL),
+                        CheckpointType.CHECKPOINT,
```

Comment:
I think you should make the checkpoint type configurable.
Moreover, I think the correct checkpoint type here should be `CheckpointType.FULL_CHECKPOINT`, please refer [FLINK-25192] for more details.

Reply:
THX for your comments. I will make it configurable. 
And CheckpointType.FULL_CHECKPOINT is clearer type in new flink version.